### ISSUE TITLE
It is not possible to rename a project with 20 characters exactly

### ISSUE THREAD

**Issue by gcoutable** [eclipse-sirius/sirius-components#665]
* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- latest
* [x] **I have specified my environment** : 
- my computer: Windows
- Cloud

### Screenshots

![Rename-project-error](https://user-images.githubusercontent.com/7371042/127467470-a63fee39-18fa-421d-86fc-2ae08f304f73.png)

### Steps to reproduce

1. Create a project
2. From the project browser try to rename the project with "TE-192-renamed-admin"

### Expected behavior

There is two possible expected behavior:
- The project should be renamed if we consider the maximum of possible characters as valid (The frontend say so), and thus, the server should not return an error
- The server should return an error (as it currently does), but the dialog should be invalid

To summarize, the frontend and the backend should be consistent.

### Actual behavior

The dialog is valid, but the server returns an error (see screenshot)

**Comment by sbegaudeau** [comment]
Closing this issue since it is not related to Sirius Components
